Gems/AtomLyIntegration/EMotionFXAtom/Code/Source/AtomActorDebugDraw.cpp

```diff
@@ -1084,30 +1084,30 @@ namespace AZ::Render
     // Find the trajectory path for a given actor instance
     AtomActorDebugDraw::TrajectoryTracePath* AtomActorDebugDraw::FindTrajectoryPath(const EMotionFX::ActorInstance* actorInstance)
     {
-        for (TrajectoryTracePath* trajectoryPath : m_trajectoryTracePaths)
+        for (const auto& trajectoryPath : m_trajectoryTracePaths)
         {
             if (trajectoryPath->m_actorInstance == actorInstance)
             {
-                return trajectoryPath;
+                return trajectoryPath.get();
             }
         }
 
         // We haven't created a path for the given actor instance yet, do so
-        TrajectoryTracePath* tracePath = new TrajectoryTracePath();
+        AZStd::unique_ptr<TrajectoryTracePath> tracePath = AZStd::make_unique<TrajectoryTracePath>();
```

Comment:
You could choose to use `auto` here instead of spelling out the type twice. But it's not a big deal.

Reply:
done.